components/Navbar.jsx

```diff
@@ -0,0 +1,105 @@
+import React, { useState } from 'react';
+import { SafeAreaView, StatusBar, StyleSheet, Text, View, TouchableOpacity } from 'react-native';
+
+const Navbar = ({ onColorChange }) => {
+  const [activeIndex, setActiveIndex] = useState(null);
+
+  const colors = ['#FFD3FA', '#FFE2CC', '#FFF7CC', '#D3F2D7', '#CDF6FF'];
+
+  const handleCirclePress = (index) => {
+    setActiveIndex(index);
+    onColorChange(colors[index]);
+  };
+
+  return (
+    <SafeAreaView style={styles.navbarContainer}>
+      <StatusBar barStyle="dark-content" />
+      <View style={[styles.navbar, activeIndex !== null && { backgroundColor: colors[activeIndex] }]}>
+        <Text style={styles.areaHeading}>
+          <Text style={styles.text}>obszar: </Text>
+          <Text style={styles.highlight}>OBSZAR</Text>
+        </Text>
+      </View>
+      <View style={styles.circleContainer}>
+        {['c', 'u', 'r', 'e', 'd'].map((letter, index) => (
+          <TouchableOpacity key={index} onPress={() => handleCirclePress(index)}>
+            <View style={[styles.circle, styles[`circle${index}`], activeIndex === index && styles.activeCircle]}>
+              <Text style={[styles.circleText, activeIndex === index && styles.activeCircleText]}>{letter}</Text>
+            </View>
+          </TouchableOpacity>
+        ))}
+      </View>
+    </SafeAreaView>
+  );
+};
+
+const styles = StyleSheet.create({
+  navbarContainer: {
+    display: 'flex',
+    alignItems: 'center',
+    backgroundColor:'white',
+  },
+  navbar: {
+    height: 120,
+    width: '100%',
+    justifyContent: 'center',
+  },
+  text: {
+    fontSize: 20,
+    fontWeight: '300',
+    color: '#535353',
+  },
+  highlight: {
+    fontSize: 25,
+    fontWeight: '600',
+    color: '#535353',
+    letterSpacing: 3,
+  },
+  areaHeading: {
+    marginLeft: 25,
+  },
+  circleContainer: {
+    display: 'flex',
+    justifyContent: 'space-around',
+    flexDirection: 'row',
+    width: '95%',
+    marginTop: -25,
+  },
+  circle: {
+    width: 45,
+    height: 43,
+    borderRadius: 100,
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  circleText: {
+    fontSize: 19,
+    fontWeight: '300',
+    color: '#000',
+  },
+  circle0: {
+    backgroundColor: '#FFD3FA',
+  },
+  circle1: {
+    backgroundColor: '#FFE2CC',
+  },
+  circle2: {
+    backgroundColor: '#FFF7CC',
+  },
+  circle3: {
+    backgroundColor: '#D3F2D7',
+  },
+  circle4: {
+    backgroundColor: '#CDF6FF',
+  },
```

Comment:
Also here move colors to constants.

Also I think circle+number is not much readible, I'd say that implementing color setting directly in components instead of setting it in styles would be better, but feel free to disagree